test/extension/handlers/topologymutation/handler_integration_test.go

```diff
@@ -0,0 +1,486 @@
+			ClusterNetwork: &clusterv1.ClusterNetwork{
+				Pods: &clusterv1.NetworkRanges{
+					CIDRBlocks: []string{"192.168.0.0/16"},
+				},
+			},
+			Topology: &clusterv1.Topology{
+				Version: "v1.29.0",
+				// NOTE: Class name must match the ClusterClass name.
+				Class: "quick-start-runtimesdk",
+				ControlPlane: clusterv1.ControlPlaneTopology{
+					Replicas: ptr.To[int32](1),
+				},
+				Workers: &clusterv1.WorkersTopology{
+					MachineDeployments: []clusterv1.MachineDeploymentTopology{
+						{
+							Name: "md-test1",
+							// NOTE: MachineDeploymentClass name must match what is defined in ClusterClass packages.
+							Class:    "default-worker",
+							Replicas: ptr.To[int32](1),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createClusterClassFakeClientAndManager(blueprint *scope.ClusterBlueprint) (client.Client, manager.Manager, error) {
+	scheme := runtime.NewScheme()
+	_ = clusterv1.AddToScheme(scheme)
+	_ = apiextensionsv1.AddToScheme(scheme)
+	objs := []client.Object{
+		blueprint.ClusterClass,
+	}
+
+	unstructuredObjs := []*unstructured.Unstructured{
+		blueprint.InfrastructureClusterTemplate,
+		blueprint.ControlPlane.Template,
+		blueprint.ControlPlane.InfrastructureMachineTemplate,
+	}
+	for _, md := range blueprint.MachineDeployments {
+		unstructuredObjs = append(unstructuredObjs, md.InfrastructureMachineTemplate, md.BootstrapTemplate)
+	}
+	for _, mp := range blueprint.MachinePools {
+		unstructuredObjs = append(unstructuredObjs, mp.InfrastructureMachinePoolTemplate, mp.BootstrapTemplate)
+	}
+
+	objAlreadyAdded := sets.Set[string]{}
+	crdAlreadyAdded := sets.Set[string]{}
+	for _, unstructuredObj := range unstructuredObjs {
+		if !objAlreadyAdded.Has(unstructuredObj.GroupVersionKind().Kind + "/" + unstructuredObj.GetName()) {
+			objs = append(objs, unstructuredObj)
+			objAlreadyAdded.Insert(unstructuredObj.GroupVersionKind().Kind + "/" + unstructuredObj.GetName())
+		}
+
+		crd := generateCRDForUnstructured(unstructuredObj)
+		if !crdAlreadyAdded.Has(crd.Name) {
+			objs = append(objs, crd)
+			crdAlreadyAdded.Insert(crd.Name)
+		}
+	}
+
+	fakeClient := fake.NewClientBuilder().WithScheme(scheme).WithObjects(objs...).WithStatusSubresource(&clusterv1.ClusterClass{}).Build()
+
+	mgr, err := ctrl.NewManager(&rest.Config{}, ctrl.Options{Scheme: scheme})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return fakeClient, mgr, err
+}
+
+func generateCRDForUnstructured(u *unstructured.Unstructured) *apiextensionsv1.CustomResourceDefinition {
+	gvk := u.GroupVersionKind()
+	return &apiextensionsv1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: contract.CalculateCRDName(gvk.Group, gvk.Kind),
+			Labels: map[string]string{
+				clusterv1.GroupVersion.String(): gvk.Version,
+			},
+		},
+	}
+}
+
+// getScope gets blueprint (ClusterClass) and current state based on cluster and clusterClassFile.
+// Note: MachinePools have not been implemented as they are not supported by CAPV.
```

Comment:
MachinePools are there in the test and this is not CAPV :-) .


```suggestion
```

Reply:
Yeah I noticed that they were not supported because I had to implement them. Forgot the note though 🗡️ 
Thx!